port/common/omrgetsysname.c

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright (c) 2021, 2021 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+/**
+ * @file
+ * @ingroup Port
+ * @brief shared library
+ */
+#include <string.h>
+#include "omrport.h"
+#include "omrgetsysname.h"
+
+#define SYSNAME_STRING "%sysname"
+#define SYSNAME_STRING_LENGTH sizeof(SYSNAME_STRING)
+
+/* Generic version of omrget_sysname() */
+uintptr_t
+omrget_sysname(struct OMRPortLibrary *portLibrary, char *sysname, uintptr_t length)
+{
+	uintptr_t result = 0;
+
+	/* Check that caller provided enough space for the string. */
+	if ((NULL == sysname) || (length < SYSNAME_STRING_LENGTH)) {
+		result = SYSNAME_STRING_LENGTH;
+	} else {
+		/* Default behaviour for platforms other than zOS, simply return the sysname string token. */
+		strcpy(sysname, SYSNAME_STRING);
```

Comment:
The header file says 'an empty string is returned if the SYSNAME is not available', contrary to what's happening here.

Reply:
Updated the function comment.